bug: pytest plugin does not support the usage of relative imports

### Component

Python SDK

### Infrahub SDK version

1.1.0

### Current Behavior

You are unable to use the pytest plugin for user provided resources that use a relative import.

```bash
tests/test_resources.yml::infrahub_check__check_tag__smoke-test FAILED                                                 [100%]

========================================================== FAILURES ==========================================================
______________________________________ resource: infrahub_check__check_tag__smoke-test _______________________________________
attempted relative import with no known parent package
================================================== short test summary info ===================================================
FAILED tests/test_resources.yml::infrahub_check__check_tag__smoke-test
===================================================== 1 failed in 0.03s ======================================================

```

### Expected Behavior

The pytest plugin works for user provided resources that use relative imports

### Steps to Reproduce

- spin up an instance of Infrahub
- create a local repository and initialize it as a repository
- create the file `check_tag.py` in the directory
```python
from infrahub_sdk.checks import InfrahubCheck
from .utils import something

class MyCheck(InfrahubCheck):
    query = "check_tag"

    async def validate(self, data: dict) -> None:
        a = something()
        return
```
- create the file `utils.py` in the directory
```python
def something() -> str:
    return "something"
```
- create the file `check_tag.gql` in the directory
```graphql
query check_tag($name: String!) {
  BuiltinTag(name__value: $name) {
    edges {
      node {
	id
      }
    }
  }
}
```
- create the file `.infrahub.yml` in the directory
```yaml
---
check_definitions:
  - name: check_tag
    class_name: MyCheck
    file_path: "check_tag.py"
    targets: tags
    parameters:
      name: "name__value"
```
- create a tests directory in the repository
- create a file `tests/test_resources.yml`
```yaml
---
version: "1.0"
infrahub_tests:
  - resource: Check
    resource_name: check_tag
    tests:
      - name: smoke-test
        expect: PASS
        spec:
          kind: check-smoke
```
- in the root of the respository execute the following command
```bash
pytest -v
```

### Additional Information

_No response_

--- a/changelog/166.fixed.md
+++ b/changelog/166.fixed.md
@@ -1,0 +1,1 @@
+Fix relative imports for the pytest plugin, note that the relative imports can't be at the top level of the repository alongside .infrahub.yml. They have to be located within a subfolder.

--- a/infrahub_sdk/_importer.py
+++ b/infrahub_sdk/_importer.py
@@ -16,6 +16,13 @@ module_mtime_cache: dict[str, float] = {}
 def import_module(
     module_path: Path, import_root: Optional[str] = None, relative_path: Optional[str] = None
 ) -> ModuleType:
+    """Imports a python module.
+
+    Args:
+        module_path (Path): Absolute path of the module to import.
+        import_root (Optional[str]): Absolute string path to the current repository.
+        relative_path (Optional[str]): Relative string path between module_path and import_root.
+    """
     import_root = import_root or str(module_path.parent)
 
     file_on_disk = module_path

--- a/infrahub_sdk/pytest_plugin/items/base.py
+++ b/infrahub_sdk/pytest_plugin/items/base.py
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import difflib
+from pathlib import Path
 from typing import TYPE_CHECKING, Any, Optional, Union
 
 import pytest
@@ -11,10 +12,10 @@ from ..exceptions import InvalidResourceConfigError
 from ..models import InfrahubInputOutputTest
 
 if TYPE_CHECKING:
-    from pathlib import Path
-
     from ...schema.repository import InfrahubRepositoryConfigElement
     from ..models import InfrahubTest
+
+_infrahub_config_path_attribute = "infrahub_config_path"
 
 
 class InfrahubItem(pytest.Item):
@@ -74,3 +75,16 @@ class InfrahubItem(pytest.Item):
 
     def reportinfo(self) -> tuple[Union[Path, str], Optional[int], str]:
         return self.path, 0, f"resource: {self.name}"
+
+    @property
+    def repository_base(self) -> str:
+        """Return the path to the root of the repository
+
+        This will be an absolute path if --infrahub-config-path is an absolut path as happens when
+        tests are started from within Infrahub server.
+        """
+        config_path: Path = getattr(self.session, _infrahub_config_path_attribute)
+        if config_path.is_absolute():
+            return str(config_path.parent)
+
+        return str(Path.cwd())

--- a/infrahub_sdk/pytest_plugin/items/check.py
+++ b/infrahub_sdk/pytest_plugin/items/check.py
@@ -1,12 +1,12 @@
 from __future__ import annotations
 
 import asyncio
+from pathlib import Path
 from typing import TYPE_CHECKING, Any, Optional
 
 import ujson
 from httpx import HTTPStatusError
 
-from ...checks import get_check_class_instance
 from ..exceptions import CheckDefinitionError, CheckResultError
 from ..models import InfrahubTestExpectedResult
 from .base import InfrahubItem
@@ -33,9 +33,12 @@ class InfrahubCheckItem(InfrahubItem):
         self.check_instance: InfrahubCheck
 
     def instantiate_check(self) -> None:
-        self.check_instance = get_check_class_instance(
-            check_config=self.resource_config,  # type: ignore[arg-type]
-            search_path=self.session.infrahub_config_path.parent,  # type: ignore[attr-defined]
+        relative_path = (
+            str(self.resource_config.file_path.parent) if self.resource_config.file_path.parent != Path() else None  # type: ignore[attr-defined]
+        )
+
+        self.check_instance = self.resource_config.load_class(  # type: ignore[attr-defined]
+            import_root=self.repository_base, relative_path=relative_path
         )
 
     def run_check(self, variables: dict[str, Any]) -> Any:

--- a/infrahub_sdk/pytest_plugin/items/python_transform.py
+++ b/infrahub_sdk/pytest_plugin/items/python_transform.py
@@ -1,12 +1,12 @@
 from __future__ import annotations
 
 import asyncio
+from pathlib import Path
 from typing import TYPE_CHECKING, Any, Optional
 
 import ujson
 from httpx import HTTPStatusError
 
-from ...transforms import get_transform_class_instance
 from ..exceptions import OutputMatchError, PythonTransformDefinitionError
 from ..models import InfrahubTestExpectedResult
 from .base import InfrahubItem
@@ -33,9 +33,11 @@ class InfrahubPythonTransformItem(InfrahubItem):
         self.transform_instance: InfrahubTransform
 
     def instantiate_transform(self) -> None:
-        self.transform_instance = get_transform_class_instance(
-            transform_config=self.resource_config,  # type: ignore[arg-type]
-            search_path=self.session.infrahub_config_path.parent,  # type: ignore[attr-defined]
+        relative_path = (
+            str(self.resource_config.file_path.parent) if self.resource_config.file_path.parent != Path() else None  # type: ignore[attr-defined]
+        )
+        self.transform_instance = self.resource_config.load_class(  # type: ignore[attr-defined]
+            import_root=self.repository_base, relative_path=relative_path
         )
 
     def run_transform(self, variables: dict[str, Any]) -> Any:

